### PR TITLE
Bug 1727800: Improve OperatorHub Load Time

### PR DESCRIPTION
**Pull request by alecmerdler**
### Description

Consume the `packagemanifests/icon` subresource to fetch Operator package icons, instead of the base64-encoded data in the `PackageManifest` response. The tiles will be rendered quickly using the `PackageManifest` response, and the icons will load in afterwards and be cached effectively using ETags.

The next stage of this will be to remove the base64 data from the `PackageManifest` object (in the `operator-lifecycle-manager` repository), reducing the response size and significantly improving the load time for **OperatorHub** without needing any further changes to `openshift/console` codebase.

Depends on https://jira.coreos.com/browse/OLM-1211
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1727800

### CODE REVIEW

File: frontend/public/components/operator-hub/operator-hub-page.tsx [hunk 1]
```diff
@@ -34,7 +32,7 @@ export const OperatorHubList: React.SFC<OperatorHubListProps> = (props) => {
       subscription: subscriptionFor(subscription.data)(operatorGroup.data)(pkg.status.packageName)(namespace),
       // FIXME: Just use `installed`
       installState: installedFor(subscription.data)(operatorGroup.data)(pkg.status.packageName)(namespace) ? 'Installed' : 'Not Installed',
-      imgUrl: iconObj ? `data:${iconObj.mediatype};base64,${iconObj.base64data}` : operatorImg,
```
Comment (alecmerdler) [inline frontend/public/components/operator-hub/operator-hub-page.tsx]: The `CatalogTile` component from `patternfly-react` [doesn't support icon image fallback](https://github.com/patternfly/patternfly-react/issues/2739).

File: frontend/public/components/operator-hub/operator-hub-page.tsx [hunk 2]
```diff
@@ -14,16 +14,14 @@ import { PackageManifestKind, OperatorGroupKind, SubscriptionKind } from '../ope
...
-
 export const OperatorHubList: React.SFC<OperatorHubListProps> = (props) => {
   const {operatorGroup, subscription, loaded, loadError, namespace = ''} = props;
   const marketplaceItems = _.get(props.marketplacePackageManifest, 'data', [] as PackageManifestKind[]);
   const localItems = _.get(props.packageManifest, 'data', [] as PackageManifestKind[]);
   const items = marketplaceItems.concat(localItems).map(pkg => {
     const currentCSVDesc = _.get(pkg, 'status.channels[0].currentCSVDesc', {});
     const currentCSVAnnotations: OperatorHubCSVAnnotations = _.get(currentCSVDesc, 'annotations', {});
-    const iconObj = _.get(currentCSVDesc, 'icon[0]');
+    const {name, currentCSV} = pkg.status.channels[0];
```
Comment (spadgett) [inline frontend/public/components/operator-hub/operator-hub-page.tsx]: Will `status.channels` always be set? I see we use `_.get` above

Comment (spadgett) [reply frontend/public/components/operator-hub/operator-hub-page.tsx]: If we're going to destructure, we should add `currentCSVDesc` from the `_.get` call above

Comment (alecmerdler) [reply frontend/public/components/operator-hub/operator-hub-page.tsx]: Done.

**Comment by alecmerdler** [comment]
@spadgett All green, ready for review.

**Pull request merged**